harness/determined/cli/experiment.py

```diff
@@ -578,6 +578,30 @@ def experiment_logs(args: Namespace) -> None:
         )
 
 
+def set_log_retention(args: Namespace) -> None:
+    if not args.forever and not args.days:
+        raise cli.errors.CliError(
+            "Please provide an argument to set log retention. --days sets the number of days to"
+            + " retain logs from the time of creation, eg. `det e set log-retention 1 "
+            + "--days 50`. --forever retains logs indefinitely, eg.`det e set log-retention "
+            + "--days none --forever."
+        )
+    elif args.forever and args.days:
```

Comment:
I don't mind this solution. Looks like @azhou-determined is also on board. Minor nitpick on the error text:
`det e set log-retention  "--days none --forever` -> `det e set log-retention 1 --forever`

I'd go as far as to type check `args.days`. 
so 
`elif args.forever and args.days` -> `elif args.forever and isinstance(args.days, int)`

Reply:
done!